### DISCUSSION TITLE
Question: On setting title for notes.

### DISCUSSION THREAD

**Discussion by MadFisa** [emacs-citar/citar-org-roam#46] [Q&A]
Hi,
I am a recently converted Vim user switching to doom emacs and a complete noob. I believe this wonderful plugin is part of the biblio module of doom and is trying to configure it to my needs. 
In particular, I have been trying to set up a template for roam note title using 

`(setq citar-org-roam-note-title-template`

but I am unable to do so. I want the title be of the format " Author et al year". I tried using
`(setq citar-org-roam-note-title-template "${author:%etal} - ${year:4}")`

But it did not work as expected, instead giving a blank for the author and sometimes giving a blank for year too. What is the required configuration to achieve this?

Similarly, I have tried setting up template using the example given in the readme for `org-roam-capture-templates`,
It prompts me citar-citekey,citar-date etc. I believe this is similar to #24 , at least to my limited understanding. How can I get this to work?

Also, what are the full range of variables available for this templates? I am unable to find the documentation for these, but it could just be that I am looking at the wrong places, being new to the emacs.

I am pretty much using a doom config that is close to default.

**Comment by bdarcus** [comment]
Just quickly responding, try prefacing the variable names with `citar-`, as in the README?

The variable names are defined in a defcustom, so you can confirm what they should be, or change them.

PS - I moved this to discussions.

**Reply by bdarcus** [reply]
Oh wait: not sure if Henrik merged the PR I submitted awhile back.

https://github.com/doomemacs/doomemacs/pull/7168

That advice only applies if you're using latest commits; it won't work without them.

**Reply by MadFisa** [reply]
I had tried prefacing with citar as mentioned in the README. It did not work, just gave me blank  spaces. This is what I used,
```
(after! citar-org-roam
(setq citar-org-roam-note-title-template "${citar-author}- ${citar-date}")
)
```

> Similarly, I have tried setting up template using the example given in the readme for org-roam-capture-templates,
> It prompts me citar-citekey,citar-date etc.

Coupled with this behaviour, is it because the variables are not being loaded?

> That advice only applies if you're using latest commits; it won't work without them.

I am using whateever is the latest doom commit, but it looks like the PR you submitted has not been merged yet. How are you supposed to enable icons with doom if the PR goes through? by adding +icons to biblio module?

**Reply by MadFisa** [reply]
update: adding +icons to biblio module didn't do anything

**Reply by bdarcus** [reply]
> How are you supposed to enable icons with doom if the PR goes through? by adding +icons to biblio module?

Yes.

**Reply by MadFisa** [reply]
I tried searching for `citar-author` in emacs `find-variable`, but couldn't find anything. Is this relevant?

Also would adding the lines from merged PR into config.el and trying again help?

**Reply by MadFisa** [reply]
Also the pinned version of citar-org-roam in doom emacs is an older version at  `86e9861a4931a4a0a4a7fc6c2be153ceb937a869`

**Reply by bdarcus** [reply]
Right; that's why I mentioned the yet-to-be-merged Doom PR.

**Reply by MadFisa** [reply]
Sorry, I misunderstood. I thought you were talking about doom emacs version. Let me try and unpin and see what happens.

**Comment by MadFisa** [comment]
Unpinning didn't change anything. Still gives blank on prefixing the variables with `citar-`.
Also was trying to get icons work with citar using instructions at https://github.com/emacs-citar/citar/wiki/Indicators, which gave me an error :
`Error caused by user's config or system: /home/user/.config/doom/config.el, (void-function citar-indicator-create)`

**Reply by bdarcus** [reply]
That can be caused by load order issues.

You might try just downloading the files in the PR to your local Doom modules dir?

Call it `mybib` and use that instead?

**Reply by MadFisa** [reply]
Didn't work. I am running out of ideas to try  😕

**Reply by bdarcus** [reply]
What does your `config.el` look like with that (it should really only set paths), and what error(s) do you get?

**Reply by MadFisa** [reply]
You can see my doom configs here 
https://github.com/MadFisa/dotfiles/blob/master/doom/config.el

I do not get any errors except the one I mentioned above that I get when I try to add icons (i.e th void function one).

Do I have to run emacs with --debug-init flag to get more details?

**Reply by bdarcus** [reply]
Just to update, the doom biblio module PR was merged a few weeks ago. If you haven't already updated it, you should do so.
